src/lp_data/HighsLp.cpp

```diff
@@ -594,14 +594,13 @@ void HighsNameHash::form(const std::vector<std::string>& name) {
   size_t num_name = name.size();
   this->clear();
   for (size_t index = 0; index < num_name; index++) {
-    const bool duplicate = !this->name2index.emplace(name[index], index).second;
+    auto emplace_result = this->name2index.emplace(name[index], index);
+    const bool duplicate = !emplace_result.second;
     if (duplicate) {
       // Find the original and mark it as duplicate
-      auto search = this->name2index.find(name[index]);
-      assert(search != this->name2index.end());
+      auto& search = emplace_result.first;
       assert(int(search->second) < int(this->name2index.size()));
-      this->name2index.erase(search);
-      this->name2index.insert({name[index], kHashIsDuplicate});
+      search->second = kHashIsDuplicate;
```

Comment:
By  deleting

>      this->name2index.erase(search);
      this->name2index.insert({name[index], kHashIsDuplicate});

and only setting 

> search->second = kHashIsDuplicate;

(which has no effect since search is immediately destroyed) the property of marking duplicate names is lost. hence errors due to the same name being given to different columns are not trapped.

Reply:
`emplace` will only insert when there is no duplication. When there is duplication, `search` points to the duplicated key and value, and setting the value will influence the original map.

This is an example: https://godbolt.org/z/YsYYrKrPb

Reply:
Thanks, it did cross my mind that there was more to setting 

> search->second = kHashIsDuplicate;

than I gave to credit for, sorry.

As I said, I'm not fluent in the use of `std::unordered_map` :-)

Reply:
Never mind. I also learned about the meaning of return values of `emplace` today. It is a clever choice to use `emplace` to detect duplication.